shopping_cart/features/step_definitions.rb

```diff
@@ -33,30 +33,40 @@
 end
 
 Then "website should redirect to '$item' page" do |item|
-  expect(page).to have_content "'#{item}!'"
+  expect(page).to have_content "Welcome to '#{item}''s page!"
 end
 
 Given "the cart total is '$amount'" do |amount|
   MyShoppingCartApp.cart.total = amount
 end
 
-When "coupon amount is '$amount'" do |amount|
+When "coupon amount '$amount' is applied" do |amount|
   MyShoppingCartApp.cart.coupon_total = amount
 end
 
 When "entering address '$address'" do |address|
   visit '/'
-  form = find '.shipping_address'
-  within(form) do
+  within('.shipping_address') do
     fill_in 'address', with: address
     click_button 'Submit'
   end
 end
 
-Then "shipping price should be '$amount' dollars" do |amount|
-  expect(MyShoppingCartApp.cart.shipping_price).to_eq(amount)
+When "entering nil address" do
+  visit '/'
+  within('.shipping_address') do
+    click_button 'Submit'
+  end
+end
+
+Then "shipping price should be '$status'" do |status|
+  expect(MyShoppingCartApp.cart.shipping_price).to_be(status)
 end
 
 Then "the total cart amount should be '$amount'" do |amount|
   expect(MyShoppingCartApp.cart.total).to_eq(amount)
 end
+
+Given "the cart has one '$item'" do |item|
+  MyShoppingCartApp.cart.item_amounts = { item => 1 }
```

Comment:
The "one" looks like it's wanting to be a dynamic value. Maybe convert it to a variable (example: `'$num $items'` => the cart has '1 banana')